signature/policy_eval_signedby.go

```diff
@@ -70,6 +71,13 @@ func (pr *prSignedBy) isSignatureAuthorAccepted(image types.UnparsedImage, sig [
 			return PolicyRequirementError(fmt.Sprintf("Signature by key %s is not accepted", keyIdentity))
 		},
 		validateSignedDockerReference: func(ref string) error {
+			r := image.Reference().DockerReference()
+			if _, isCanonical := r.(reference.Canonical); isCanonical || reference.IsNameOnly(r) {
+				if !pr.UntaggedReference.matchesDockerReference(image, ref) {
```

Comment:
> See why I'm saying matchAnyTag is confusing? I would rather have something which actually says "matchAnyTagButMatchDigestPreciselyOfCourse" (joking but probably you get my point).

We don’t explicitly call out that digests are being verified anywhere else in the UI; it is kind of implied that yes the signature actually applies to the image and that the users don’t have to think about how the two are linked.

`signedIdentity` has `matchExact`/`matchRepository`, where `matchRepository` is exactly `matchAnyTag`. Perhaps we can use these names? I don’t know; on one hand it is more explicit to say what _is_ being matched (and, in a sense, reassuring to use the positive sense that _something_ is being matched), OTOH `AnyTag` is very explicit about what is the difference, whereas detailed understanding is needed to understand how `matchExact` and `matchRepository` differ.

> This is tricky then :( I guess we can still call it UntaggedReference and thoroughly document what it really means.

Per IRC, perhaps `referencesByDigest` would be both more explicit and clearer?